src/lib/chips/chip-input.ts

```diff
@@ -19,12 +38,13 @@ export interface MdChipInputEvent {
 @Directive({
   selector: 'input[mdChipInputFor], input[matChipInputFor]',
   host: {
-    'class': 'mat-chip-input',
+    'class': 'mat-chip-input mat-input-element',
     '(keydown)': '_keydown($event)',
-    '(blur)': '_blur()'
+    '(blur)': '_blur()',
+    '(focus)': '_focus()',
   }
 })
-export class MdChipInput {
+export class MdChipInput extends MdInput {
```

Comment:
I think it's okay to use something like `<input mdInput mdChipInputFor="chips">`.

Reply:
They won't conflict with each other since this extends `MdInput`? It seems like at the very least you would be doing twice the amount of work for no reason, since both directives would react to every change in value and try to do things

Reply:
@tinayuangao still curious about why this needs to extend `MdInput`, seems unnecessary to me.

Reply:
@mmalerba Changed `MdChipInput` to not extends `MdInput`. Added `focus()`, `focused` and `empty` to work with chip list. Please take another look. Thanks!